packages/airnode-adapter/src/request-building/authentication.ts

```diff
@@ -88,7 +88,19 @@ function getSchemeAuthentication(
   credentials: BaseApiCredentials | null,
   options: CachedBuildRequestOptions
 ): Partial<Authentication> {
-  switch (apiSecurityScheme.type) {
+  // TODO: The ApiSecurityScheme type imported from OIS is typed as "any" by TypeScript making the switch statement
+  // below be non-exhaustive and generate a TS error. This is a workaround.
+  type ApiSecuritySchemeType =
+    | 'apiKey'
+    | 'http'
+    | 'relayChainId'
+    | 'relayChainType'
+    | 'relayRequesterAddress'
+    | 'relaySponsorAddress'
+    | 'relaySponsorWalletAddress'
+    | 'relayRequestId';
```

Comment:
I am pretty sure there is something wrong with the `@api/ois` v2.3.0 release... 😕 

What I did:
- Confirmed the type is exported correctly in the latest airnode master (which uses @api3/ois v2.2.1)
- Checked out this PR, installed & built packages, and confirmed that the type is indeed now `any`
- Replaced `"@api3/ois": "2.3.0",` with `"@api3/ois": "file:../../../ois",` (which points to the latest master built locally) in `package.json` of `airnode-adapter`, reinstalled, and was able to delete this addition because the type is correct.
- Confirmed that `ApiSecurityScheme` in `dist/src/ois.d.ts` is `any` in `@api/ois` v2.3.0, but is correctly typed in that file when `@api3/ois` is built locally from the current master 

**Most concrete proof**:
- Published OIS v2.3.0 myself as https://www.npmjs.com/package/test-ois
- Replaced `@api3/ois` with `test-ois` and the typing is correct

Reply:
> I am pretty sure there is something wrong with the @api/ois v2.3.0 release... 😕

That is pretty sad and also bad. I followed the release instructions (with the exception of pushing to main) so I don't think I made a mistake. I think that the files built on my machine somehow differ with yours...

Reply:
> Published OIS v2.3.0 myself

Did you do it from the v2.3.0 commit?

Reply:
I published `2.3.0` of `test-ois` from the latest `main` (which includes 2 commits since v2.3.0). To see if those commits somehow had something to do with it, I just now checked out the v2.3.0 commit (https://github.com/api3dao/ois/commit/7d00c0a62bb5f28f5dc0b3601ecd7967773c2ea7), rebuilt, and [published that](https://www.npmjs.com/package/test-ois?activeTab=versions) as `2.3.0-alpha` of `test-ois`. I then used that as an Airnode dependency and same result- typing works.

So I'm not sure why your build appears to have differed, but I'd like to publish v2.3.1 of OIS to avoid hacking the typing here.

Reply:
I troubleshoot this and I think I understand what happened. 

TLDR: We can just publish a 2.3.1 patch from main and it's going to work.

What happened:
1. I correctly published 2.3.0 but the typings are broken in Airnode (this PR)
2. In the meantime, you merged two dependency updates, including the Typescript dependency in OIS.
3. [Here](https://github.com/api3dao/airnode/pull/1914#discussion_r1408839509) you tried using the locally built OIS and the typings are correct (suggesting issue with 2.3.0 release).
4. [Here](https://github.com/api3dao/airnode/pull/1914#discussion_r1409553353) you've published `2.3.0-alpha` but you've actually not rebuilt that (you can verify that when you install the dependency and compare package.json with dist/package.json) so actually both `test-ois@2.3.0` and `test-ois@2.3.0-alpha` are "same" and both work.
5. I built the OIS myself, compared it deeply with `test-ois@2.3.0-alpha` and the content was the same (apart from discrepancies in the package.json).

Reply:
> TLDR: We can just publish a 2.3.1 patch from main and it's going to work.

Agree.

The rest is partially correct with a big caveat at the end. As a final step I first checked out https://github.com/api3dao/ois/commit/7d00c0a62bb5f28f5dc0b3601ecd7967773c2ea7, which is the v2.3.0 commit before the additional typescript and lodash dependabot commits (I didn't merge those btw). I then ran `yarn install && yarn build`, then released [test-ois@2.3.0-beta](https://www.npmjs.com/package/test-ois/v/2.3.0-beta?activeTab=versions). I then did a recursive diff within `dist` and it does differ from `test-ois@2.3.0`, which was built from master, as well as `test-ois@2.3.0-alpha`, which you correctly point out wasn't fully rebuilt before publishing and therefore is the same as `test-ois@2.3.0`.

However, probably the most important part is that `test-ois@2.3.0-beta` works in Airnode with the correct types, which points to something having gone wrong in your build and release of v2.3.0 (what exactly I have no clue).

Anyway, to me this is even more motivation for https://github.com/api3dao/ois/issues/122

Reply:
Just released v2.3.1 on npm and [GitHub](https://github.com/api3dao/ois/releases/tag/v2.3.1). As we expected, it looks to have solved this typing issue. Just remember to now bump & rerun the `update-ois-version.ts` script.

Reply:
> However, probably the most important part is that test-ois@2.3.0-beta works in Airnode with the correct types, which points to something having gone wrong in your build and release of v2.3.0 (what exactly I have no clue).

You are right about this. I tried building the 2.3.0. locally and indeed it differs with what's published... This has to be some TS problem or maybe I didn't run `yarn` to install fresh dependencies 🤔 .